create-embeddings/open-source-new-data.ipynb

```diff
@@ -117,37 +270,55 @@
+    "\n",
     "# Generate embedding for the search query\n",
-    "query_embedding = get_embedding(\"ocean tragedy\")\n",
-    "\n",
-    "# Sample vector search pipeline\n",
-    "pipeline = [\n",
-    "   {\n",
-    "      \"$vectorSearch\": {\n",
-    "            \"index\": \"vector_index\",\n",
-    "            \"queryVector\": query_embedding,\n",
-    "            \"path\": \"embedding\",\n",
-    "            \"exact\": True,\n",
-    "            \"limit\": 5\n",
-    "      }\n",
-    "   }, \n",
-    "   {\n",
-    "      \"$project\": {\n",
-    "         \"_id\": 0, \n",
-    "         \"text\": 1,\n",
-    "         \"score\": {\n",
-    "            \"$meta\": \"vectorSearchScore\"\n",
-    "         }\n",
-    "      }\n",
-    "   }\n",
-    "]\n",
+    "query_float32_embeddings = get_embedding(query_text, precision=\"float32\")\n",
+    "query_int8_embeddings = get_embedding(query_text, precision=\"int8\")\n",
+    "query_int1_embeddings = get_embedding(query_text, precision=\"ubinary\")\n",
+    "\n",
+    "# Convert each embedding to BSON format\n",
+    "query_bson_float32_embeddings = generate_bson_vector(query_float32_embeddings, BinaryVectorDtype.FLOAT32)\n",
+    "query_bson_int8_embeddings = generate_bson_vector(query_int8_embeddings, BinaryVectorDtype.INT8)\n",
+    "query_bson_int1_embeddings = generate_bson_vector(query_int1_embeddings, BinaryVectorDtype.PACKED_BIT)\n",
+    "\n",
+    "# Define vector search pipeline for each precision\n",
+    "pipelines = []\n",
+    "for query_embedding, path in zip(\n",
+    "    [query_bson_float32_embeddings, query_bson_int8_embeddings, query_bson_int1_embeddings],\n",
+    "    [\"BSON-Float32-Embedding\", \"BSON-Int8-Embedding\", \"BSON-Int1-Embedding\"]\n",
+    "):\n",
+    "    pipeline = [\n",
+    "       {\n",
+    "          \"$vectorSearch\": {\n",
+    "                \"index\": \"vector_index\",  # Adjust if necessary\n",
+    "                \"queryVector\": query_embedding,\n",
+    "                \"path\": path,\n",
+    "                \"exact\": True,\n",
+    "                \"limit\": 5\n",
+    "          }\n",
+    "       },\n",
+    "       {\n",
+    "          \"$project\": {\n",
+    "             \"_id\": 0,\n",
+    "             \"data\": 1,\n",
+    "             \"score\": {\n",
+    "                \"$meta\": \"vectorSearchScore\"\n",
+    "             }\n",
+    "          }\n",
+    "       }\n",
+    "    ]\n",
+    "    pipelines.append(pipeline)\n",
     "\n",
-    "# Execute the search\n",
-    "results = collection.aggregate(pipeline)\n",
+    "# Execute the search for each precision\n",
+    "for pipeline in pipelines:\n",
+    "    print(f\"Results for {pipeline[0]['$vectorSearch']['path']}:\")\n",
+    "    results = collection.aggregate(pipeline)\n",
     "\n",
-    "# Print results\n",
-    "for i in results:\n",
-    "   print(i)\n"
+    "    # Print results\n",
+    "    for i in results:\n",
+    "        print(i)"
```

Comment:
Just noting - I get the same `RuntimeWarning`s in `sentence_transformers` when I execute this cell:

> /Users/dachary.carey/Library/Python/3.9/lib/python/site-packages/sentence_transformers/quantization.py:434: RuntimeWarning: invalid value encountered in divide
  return ((embeddings - starts) / steps - 128).astype(np.int8)
/Users/dachary.carey/Library/Python/3.9/lib/python/site-packages/sentence_transformers/quantization.py:434: RuntimeWarning: invalid value encountered in cast
  return ((embeddings - starts) / steps - 128).astype(np.int8)

Reply:
The warning disappeared when I upgraded the kernel to 3.10, but returned. Still think that it's ultimately an issue with the model itself